src/middlewared/middlewared/alert/source/active_directory.py

```diff
@@ -39,6 +40,15 @@ async def check(self):
                 key=None
             )
 
+        try:
+            await self.middleware.call("activedirectory.check_nameservers")
```

Comment:
Might be missing something, but are the domain and site parameters missing from this call?

```python
Failed to check for alert ActiveDirectoryDomainHealth: Traceback (most recent call last): File "/usr/lib/python3/dist-packages/middlewared/plugins/alert.py", line 776, in __run_source alerts = (await alert_source.check()) or [] File "/usr/lib/python3/dist-packages/middlewared/alert/source/active_directory.py", line 44, in check await self.middleware.call("activedirectory.check_nameservers") File "/usr/lib/python3/dist-packages/middlewared/main.py", line 1386, in call return await self._call( File "/usr/lib/python3/dist-packages/middlewared/main.py", line 1335, in _call return await methodobj(*prepared_call.args) TypeError: check_nameservers() missing 1 required positional argument: 'domain'
```

Reply:
Doh, looks like you're right. I'll get a fix prepped. Thanks for pointing this out.

Reply:
Awesome, thanks!
The error happened with today's release of SCALE 22.12.1, but I didn't see an issue in the tracker yet. If needed, I can create one.

Reply:
https://github.com/truenas/middleware/pull/10729

Already created. NAS-120430